src/dune_file_watcher/dune_file_watcher.ml

```diff
@@ -340,12 +340,13 @@ let fswatch_backend () =
     User_error.raise [ Pp.text "Please install fswatch to enable watch mode." ]
 
 let select_watcher_backend () =
-  if Sys.linux then (
+  match Platform.OS.value with
+  | Linux ->
     assert (Ocaml_inotify.Inotify.supported_by_the_os ());
-    `Inotify_lib)
-  else if Fsevents.available () then `Fsevents
-  else if Sys.win32 then `Fswatch_win
-  else fswatch_backend ()
+    `Inotify_lib
+  | Darwin -> if Fsevents.available () then `Fsevents else `Fswatch_win
```

Comment:
That doesn't seem correct. Perhaps you should just use `with` in the match case and point to `Fsevents like it was before. The the Other case should be a catch all.

Reply:
fixed now